client/src/containers/InsertLinkModal/InsertLinkModal.js

```diff
@@ -0,0 +1,134 @@
+import React, { PropTypes, Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import FormBuilderModal from 'components/FormBuilderModal/FormBuilderModal';
+import * as schemaActions from 'state/schema/SchemaActions';
+
+class InsertLinkModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setOverrides(props.show ? props : null);
+  }
+
+  componentWillReceiveProps(props) {
+    if (props.show && !this.props.show || !props.show && this.props.show) {
+      this.setOverrides(props.show ? props : null);
+    }
+  }
+
+  componentWillUnmount() {
+    this.setOverrides();
+  }
+
+  /**
+   * Compares the current properties with received properties and determines if overrides need to be
+   * cleared or added.
+   * TODO move setting overrides to a HOC?
+   *
+   * @param {object} props
+   */
+  setOverrides(props = null) {
+    if (!props) {
+      // clear any overrides that may be in place
+      const schemaUrl = props && props.schemaUrl || this.props.schemaUrl;
+      if (schemaUrl) {
+        this.props.actions.schema.setSchemaStateOverrides(schemaUrl, null);
+      }
+    }
+    if (props && props.schemaUrl) {
+      const attrs = Object.assign({}, props.fileAttributes);
```

Comment:
this property was already named like this from InsertMediaModal
I'm hesitant to rename it here, as I feel we could move all this `setOverrides()` logic to a HOC to reduce code repetition :)